### PR TITLE
A few sea ice related issues in the NSST analysis: Issue #18

### PR DESCRIPTION
**Pull request by XuLi-NOAA**
The UFS_UTILS Issue #18 is ready for review

### CODE REVIEW

File: sorc/global_cycle.fd/read_write_data.f90 [hunk 1]
```diff
@@ -1499,4 +1501,352 @@ SUBROUTINE READ_DATA(TSFFCS,SMCFCS,SNOFCS,STCFCS, &
...
+
+  mlat_sst = kgds(3)   ! number points on longitude circle (360)
+  mlon_sst = kgds(2)   ! number points on latitude circle (720)
+
+  write(*,*) 'mlat_sst, mlon_sst : ',mlat_sst, mlon_sst
+
+  call baclose(lu_sst,iret)
+  if (iret /= 0 ) then
+     write(6,*)'get_tf_clm_dim:  ***error*** close sst file'
+     stop
```
Comment (GeorgeGayno-NOAA) [inline sorc/global_cycle.fd/read_write_data.f90]: I would use 'mpi_abort' instead of 'stop'.

File: ush/global_cycle.sh [hunk 2]
```diff
@@ -234,6 +234,7 @@ HOMEgfs=${HOMEgfs:-$BASEDIR/gfs_ver.${gfs_ver}}
 EXECgfs=${EXECgfs:-$HOMEgfs/exec}
 FIXfv3=${FIXfv3:-$HOMEgfs/fix/fix_fv3_gmted2010/$CASE}
 FIXam=${FIXam:-$HOMEgfs/fix/fix_am}
+FIXsst=${FIXsst:-/scratch4/NCEPDEV/da/noscrub/Xu.Li/data/rtgsst/fix}
```
Comment (yangfanglin) [inline ush/global_cycle.sh]: New FIXsst files need to be added to $HOMEgfs/fix/fix_am directory on all platforms

**Comment by XuLi-NOAA** [comment]
Fanglin,

Yes, and the tests look good.

Xu

File: ush/global_cycle.sh (hunk 2 above)
Comment (XuLi-NOAA) [reply ush/global_cycle.sh]: FIXsst  has been changed to be FIXam (are you looking at the newest version?)

File: sorc/global_cycle.fd/read_write_data.f90 (hunk 1 above)
Comment (XuLi-NOAA) [reply sorc/global_cycle.fd/read_write_data.f90]: OK, I will use mpi_abort instead of stop

**Comment by yangfanglin** [comment]
Xu, would you like to present your tests at either the DA meeting or the GFS.v16 biweekly technical meeting ?

Other than that, the code looks fine to me.

**Comment by GeorgeGayno-NOAA** [comment]
I am not sure your branch is up-to-date with 'develop'.  Can you do a merge to be certain?

**Comment by XuLi-NOAA** [comment]
Yes, I can do it, either one of the two meetings.

**Comment by GeorgeGayno-NOAA** [comment]
After your last update, cycle won't build:

`read_write_data.f90(1576): error #6404: This name does not have a type, and must have an explicit type.   [MPI_COMM_WORLD]
     CALL MPI_ABORT(MPI_COMM_WORLD, 111)`

You need to add `include "mpif.h"`

**Comment by XuLi-NOAA** [comment]
Yes, I just fixed it and it builds now

**Comment by GeorgeGayno-NOAA** [comment]
Xu, I ran stand-alone cycle using 20190730 data and compared your branch against 'develop'.   Using a netcdf tool, I see differences in 'tsea', 'tisfc', 'tref', 'tfinc' and 'stc'.   Is that expected?   The 'stc' differences are very large.  When viewed, the large differences are at sea ice.  Here is a summary of tile 6 differences:

```CHECK 20190730.000000.sfcanl_data.tile6.nc
Variable Group  Count         Sum      AbsSum        Min          Max     Range       Mean    StdDev
tsea     /     100014    -2186.55     2186.76   -1.63343    0.0612983   1.69473 -0.0218624  0.030319
tisfc    /      15215    -1114.58     1114.58 -0.0849977 -1.54403e-05 0.0849823 -0.0732554 0.0224935
tref     /     111665    -9272.22     9272.22  -0.267241 -1.22118e-05  0.267229  -0.083036 0.0115932
tfinc    /      11564    -781.941     781.941 -0.0849977 -1.22118e-05 0.0849855 -0.0676186 0.0250677
stc      /     461264 5.19794e+06 5.24694e+06   -8.45253      38.8503   47.3028    11.2689   9.45556
```

Why are sea ice temperatures changing so much?

**Comment by XuLi-NOAA** [comment]
George,

I have talked to Xingren and understand what happened exactly now.

As we discussed, the line,  soilt_tile(:,ij) = skint_tile(ij)  , to assign
stc needs to be removed.

I have committed to my develop fork.

Can you do the check with the statistics you did again?

Thanks for finding out this problem.

Xu

**Comment by SMoorthi-emc** [comment]
I am ok with commit
Moorthi

**Comment by GeorgeGayno-NOAA** [comment]
After the bug fix for the sea ice column temperatures, the tile 6 difference (Xu's branch minus 'develop') is:
```
CHECK 20190730.000000.sfcanl_data.tile6.nc
Variable Group  Count      Sum  AbsSum        Min          Max     Range       Mean    StdDev
tsea     /     100014 -2186.55 2186.76   -1.63343    0.0612983   1.69473 -0.0218624  0.030319
tisfc    /      15215 -1114.58 1114.58 -0.0849977 -1.54403e-05 0.0849823 -0.0732554 0.0224935
tref     /     111665 -9272.22 9272.22  -0.267241 -1.22118e-05  0.267229  -0.083036 0.0115932
tfinc    /      11564 -781.941 781.941 -0.0849977 -1.22118e-05 0.0849855 -0.0676186 0.0250677
stc      /      60860 -4458.33 4458.33 -0.0849977 -1.54403e-05 0.0849823 -0.0732554  0.022493
```
Are these reasonable?

**Comment by XuLi-NOAA** [comment]
Yes, they are reasonable and as expected.

Thanks!

**Comment by GeorgeGayno-NOAA** [comment]
Your branch is not up-to-date with 'develop'.

**Comment by GeorgeGayno-NOAA** [comment]
sfcsub.F was updated for v15.2.  The interpolation of substrate temperature was changed to not use a land mask.  Those changes are not in Xu's branch.

**Comment by GeorgeGayno-NOAA** [comment]
sfcsub.F was recently updated in the forecast model:
[see here](https://vlab.ncep.noaa.gov/redmine/projects/comfv3/repository/diff/gfsphysics/physics/sfcsub.F?utf8=%E2%9C%93&rev=01e034fe0b44bbcbafda51f1624e2bded6c1faef&rev_to=3dc7e9b1fc341bae78c4aac37582e4046a330c2d)

These updates should be incorporated into Xu's branch (in both cycle and chgres).  sfcsub.F should be consistent between the forecast model and ufs_utils.

**Comment by XuLi-NOAA** [comment]
Hi,

I thought it is already included in my fork develop.

Thanks for the remind.

It is in now.

XU

**Comment by GeorgeGayno-NOAA** [comment]
The sfcsub.F module contains a variable - 'tgice' - that is 271.2.  That variable is used in several computations.  Also, variable 'tsfomn' is 271.2 and is used to range check the lower bound of open water.  Do these need to be changed for salinity?

**Comment by XuLi-NOAA** [comment]
Hi, George,

It is a good question.

As Moothi told me, the atmospheric model is still not ready yet to use the
calculated salinity dependent water temperature for the water/ice mixed
grids, therefore, the constant value of 271.2 for tgice and tsfomn will be
used still.

The salinity dependent water temperature affects the NSST tref, and
therefore tsea (or tmpsfc), in the NSST adjust part, which is ont in
sfcsub.F.

Therefore, this time, we don't need to change  tgice and tsfomn in sfcsub.F.

Or we may need think about more to see if the  salinity dependent water
temperature can be applied to more areas, like the water/ice mixed grids
(currently, the water portion is assigned as 271.2, which is too low by
more than 1 degree for lakes with lower salinity).

XU

File: ush/global_cycle.sh [hunk 3]
```diff
@@ -58,6 +58,8 @@
 #     FNMXIC        Input maximum sea ice climatology GRIB file.
 #                   Defaults to ${FIXam}/global_maxice.2x2.grb
 #     FNTSFC        Input SST climatology GRIB file.
+#                   Defaults to ${FIXam}/global_salclm.t1534.3072.1536.nc
+#     FNSALC        Input Salinity climatology netcdf file.
```
Comment (GeorgeGayno-NOAA) [inline ush/global_cycle.sh]: There is a typo here and in the next line.

Comment (XuLi-NOAA) [reply ush/global_cycle.sh]: Fixed and pushed.
Thanks!

**Pull request merged**